homeassistant/components/alarm_control_panel/nx584.py

```diff
@@ -0,0 +1,106 @@
+"""
+Networx NX584 interface
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Configuration:
+
+To use the Example custom component you will need to add the following to
+your configuration.yaml file.
+
+alarm_control_panel:
+  platform: nx584
+  host: localhost:5007
+
+Variable:
+
+host
+*Optional
+HOST should be a something like "localhost:5007" which is the
+connection information for talking to the pynx584 backend server.
+"""
+import logging
+import requests
+
+from homeassistant.const import (STATE_UNKNOWN, STATE_ALARM_DISARMED,
+                                 STATE_ALARM_ARMED_HOME,
+                                 STATE_ALARM_ARMED_AWAY)
+import homeassistant.components.alarm_control_panel as alarm
+
+REQUIREMENTS = ['pynx584==0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Setup nx584. """
+    host = config.get('host', 'localhost:5007')
+
+    add_devices([NX584Alarm(hass, host, config.get('name', 'NX584'))])
+
+
+class NX584Alarm(alarm.AlarmControlPanel):
+    """ NX584-based alarm panel. """
+    def __init__(self, hass, host, name):
+        from nx584 import client
+        self._hass = hass
+        self._host = host
+        self._name = name
+        try:
+            self._alarm = client.Client('http://%s' % host)
+            self._alarm.list_zones()
+        except requests.exceptions.ConnectionError as ex:
+            _LOGGER.error('Unable to connect to NX584: %s',
```

Comment:
The alarm control panel component will keep polling this device because it has been added on line 38. You could consider catching the exception in `setup_platform` so that it will not get added to HA.